app/init/launch.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Linking} from 'react-native';
+import {Notifications} from 'react-native-notifications';
+
+import {Screens} from '@constants';
+import {getActiveServerCredentials, getServerCredentials} from '@init/credentials';
+import {goToScreen, resetToChannel, resetToSelectServer} from '@screens/navigation';
+import {parseDeepLink} from '@utils/url';
+
+export const initialLaunch = async () => {
+    const deepLinkUrl = await Linking.getInitialURL();
+    if (deepLinkUrl) {
+        launchAppFromDeepLink(deepLinkUrl);
+        return;
+    }
+
+    const notification = await Notifications.getInitialNotification();
+    if (notification) {
+        launchAppFromNotification(notification);
+        return;
+    }
+
+    launchApp();
+};
+
+const launchAppFromDeepLink = (deepLinkUrl: string) => {
+    const props = getLaunchPropsFromDeepLink(deepLinkUrl);
+    launchApp(props);
+};
+
+const launchAppFromNotification = (notification: NotificationWithData) => {
+    const props = getLaunchPropsFromNotification(notification);
+    launchApp(props);
+};
+
+const launchApp = async (props: OptionalLaunchProps = undefined, resetNavigation = true) => {
+    let serverUrl;
+    switch (props?.launchType) {
```

Comment:
A switch case needs to have a default case.  In this case, the default case will immediate break out.